pkg/controllers/jobset_controller.go

```diff
@@ -612,11 +612,11 @@ func constructJob(js *jobset.JobSet, rjob *jobset.ReplicatedJob, jobIdx int) (*b
 	}
 
 	// If this job should be exclusive per topology, configure the scheduling constraints accordingly.
-	if _, ok := js.Annotations[jobset.ExclusiveKey]; ok {
+	if _, ok := rjob.Template.Annotations[jobset.ExclusiveKey]; ok {
```

Comment:
To support setting it at both levels, here we should check both `rjob.Template.Annotations[jobset.ExclusiveKey]` and `js.Annotations[jobset.ExclusiveKey]`